heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesScheduler.java

```diff
@@ -253,4 +370,46 @@ protected String getFetchCommand() {
 
     return command;
   }
+
+  @Override
+  public void addContainers(Set<PackingPlan.ContainerPlan> containersToAdd) {
+    // grab the base pod so we can copy and modify some stuff
+    String basePodName = Runtime.topologyName(runtime) + "-0";
+    JsonNode podConfig;
+    try {
+      podConfig = controller.getBasePod(basePodName);
+    } catch (IOException ioe) {
+      LOG.log(Level.SEVERE, "Problem retrieving base pod configuration", ioe);
+      throw new TopologyRuntimeManagementException("Unable to add containers for update");
+    }
+
+    // iterate over the containers we need to add and rebuild the spec based on the new plan
+    ObjectMapper mapper = new ObjectMapper();
+    for (PackingPlan.ContainerPlan containerPlan : containersToAdd) {
```

Comment:
Does kubernetes provide the ability to add/remove multiple containers with one call? Doing it individually leaves us in a bad state if we fail midway through. If not we should clearly log that state upon failure (which ones succeeded, which failed or were never attempted).